go/src/koding/kites/kloud/utils/object/printer.go

```diff
@@ -0,0 +1,113 @@
+package object
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/fatih/structs"
+)
+
+type Printer struct {
+	Tag    string
+	JSON   bool
+	Format string
+	W      io.Writer
+}
+
+func (p *Printer) Print(v interface{}) error {
+	if p.JSON {
+		return p.printJSON(v)
+	}
+	return p.printTab(v)
+}
+
+func (p *Printer) printJSON(v interface{}) error {
+	enc := json.NewEncoder(p.W)
+	enc.SetIndent("", "\t")
+
+	return enc.Encode(v)
+}
+
+var (
+	fprintf  = reflect.ValueOf(fmt.Fprintf)
+	stringer = reflect.TypeOf((*fmt.Stringer)(nil)).Elem()
+)
+
+func (p *Printer) printTab(v interface{}) error {
```

Comment:
I believe we can implement the same functionality with

* encode to json 
* decode the result to map[string]interface{}
* get the keys and output them.

we can also add embedded field support easily with a recursive function.

Lets not use reflection whenever possible in our app logic.

ie: getting recursive field names
```
func getKeys(prefix string, data map[string]interface{}) {
	for key, val := range data {
		fieldKey := prefix + key
                // do your thingy with fieldKey
		next, ok := val.(map[string]interface{})
		if ok {
			getKeys(fieldKey+".", next)
		}
	}
}
```